### ISSUE TITLE
💅 `useRegexpExec` should cover `new RegExp()`

### ISSUE THREAD

**Issue by Netail** [biomejs/biome#8285]
### Environment information

```bash
-
```

### Rule name

useRegexpExec

### Playground link

https://biomejs.dev/playground/?code=JwBzAG8AbQBlAHQAaABpAG4AZwAnAC4AbQBhAHQAYwBoACgAbgBlAHcAIABSAGUAZwBFAHgAcAAoAC8AdABoAGkAbgBnAC8AKQApADsA&ruleDomains.react=recommended

### Expected result

Cover `new RegExp()` too, not only regex literals. Already included in tests

Ref https://github.com/biomejs/biome/pull/8034

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
